droidlet/tools/crowdsourcing/droidlet_static_html_task/pilot_task_run.py

```diff
@@ -0,0 +1,151 @@
+SAFETY_WORDS = get_safety_words()
+
+@dataclass
+class TestScriptConfig(RunScriptConfig):
+    defaults: List[Any] = field(default_factory=lambda: defaults)
+    task_dir: str = TASK_DIRECTORY
+
+
+register_script_config(name="scriptconfig", module=TestScriptConfig)
+
+
+def validate_answers(answers, gt):
+    freeform_answer = answers["freeform_answer"]
+    commands = freeform_answer.splitlines()
+
+    # filter empty commands
+    filtered_commands = [x for x in commands if x != ""]
+
+    # 1: Check that the number of commands >= 4
+    if len(filtered_commands) < 4:
+        return False
+
+    # 2: Length check: Check that the average number of words in commands >= 2
+    commands_split = [x.lower().split(" ") for x in filtered_commands]
+    avg_words_in_commands = sum(map(len, commands_split)) / len(commands_split)
+    if avg_words_in_commands < 2:
+        return False
+
+    # 3: Keyword matching: Check that at least 2 keywords appear in the commands
+    occurrence = sum([True if keyword in command.lower() else False for command in filtered_commands for keyword in KEYWORD_LIST])
+    if occurrence < 2:
+        return False
+
+    # 4: Safety check
+    if any([len(set(cmd) & SAFETY_WORDS) > 0 for cmd in commands_split]):
+        return False
+
+    # Validate multiple choice questions
+    if answers["answer_0"] != gt["answer_0"] or \
+       answers["answer_1"] != gt["answer_1"] or \
+       answers["answer_2"] != gt["answer_2"]:
+       return False
+
+    return True
+
+def validate_unit(unit):
+    if unit.get_assigned_agent() is None:
+        return
+    output = mephisto_data_browser.get_data_from_unit(unit)["data"]
+    if output is None:
+        return
+
+    answers = mephisto_data_browser.get_data_from_unit(unit)["data"]["outputs"]
+    worker = unit.get_assigned_agent().get_worker()
+
+    # Block every worker from working on this pilot task for second time
+    # If they pass, they will be put into an allowlist
+    # If they fail, they will never be able to see this again, unless configured explicitly
+    try:
+        db.make_qualification(PILOT_BLOCK_QUAL_NAME)
+    except:
+        pass
+    else:
+        logging.debug(f"{PILOT_BLOCK_QUAL_NAME} qualification not exists, so create one")
+    worker.grant_qualification(PILOT_BLOCK_QUAL_NAME, 1)
+    
+    # Validate pilot task answers, workers who pass the validation will be put into an allowlist
+    # by granting a qualification task to them and specify the task as a req in the full task
+    try:
+        db.make_qualification(PILOT_ALLOWLIST_QUAL_NAME)
+    except:
+        pass
+    else:
+        logging.debug(f"{PILOT_ALLOWLIST_QUAL_NAME} qualification not exists, so create one")
+
+    if validate_answers(answers, PILOT_QUAL_ANSWERS):
+        worker.grant_qualification(PILOT_ALLOWLIST_QUAL_NAME, 1)
+        logging.info(f"Worker {worker.worker_name} passed the pilot task, put him/her into allowlist")
+
+    return
```

Comment:
As it stands, this will be run on _every_ unit submitted, which may not be what you desire. 

I've built a cleaner workflow for this type of task (the [`ValidationRequired` mixin](https://github.com/facebookresearch/Mephisto/pull/566/files) and you can see an example of its usage [here](https://github.com/fairinternal/mephisto-internal/pull/18/files). Much of the boilerplate goes away when you use this, though you'll create a simple Mephisto `Blueprint` to designate that you want to use this workflow for your task.

Reply:
"As it stands, this will be run on every unit submitted, which may not be what you desire." -- may I ask why would this be a problem? Each different worker should only get to accept this task once, then he/she will be blocked from accept it again (because we grant the block qual to him/her), right?

Reply:
Ah I see right this is still set up with the two-task version. The code I linked is for a workflow to do both in one.

Reply:
Gotcha. Can I get approved for this one? Will iterate newer version later.